.github/workflows-source/ci-pr-only.yml

```diff
@@ -112,6 +112,28 @@ jobs:
         if: steps.filter.outputs.lock-generate == 'true'
         run: ./ci/scripts/lock-generate.sh
 
+  update-mainnet-artifacts:
```

Comment:
Why do we only want this to run on PRs and not the master branch?

Reply:
Because it creates commits that update `mainnet-artifacts.bzl` and pushes them to the PR. That wouldn't work on `master`. So it's very similar to the `lock-generate` job above it.

Reply:
It will run automatically on PRs touching `mainnet_revisions.json`. Which is mostly updated automatically by the [update-ic-versions-file](https://github.com/dfinity-ops/release/blob/main/.github/workflows/update-ic-versions-file.yaml) job in the `release` repo. See recent commits: 
https://github.com/dfinity/ic/commits/master/testnet/mainnet_revisions.json.

Note that `mainnet-artifacts.bzl` should always be in sync with `mainnet_revisions.json` so running it on a schedule would violate that.